### PR TITLE
UI Stubs for Smart Variables

### PR DESCRIPTION
**Pull request by jyejare**
This includes all the Smart Variable UI stubs. This feature has all the functions similar to Smart Class Parameters but has its own added/removed functions, those are following:
1. Smart Variable can be created unlike Smart class parameters.
2. Override option is not available in Smart Variables.
3. 'Use Puppet Default' option is not supported in Smart Variables.

### CODE REVIEW

File: tests/foreman/ui/test_variables.py [hunk 1]
```diff
@@ -0,0 +1,1130 @@
...
+        1. Error is displayed for invalid variable name.
+        2. The smart Variable is not created.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_delete_variable_smartVariables_menu(self):
```
Comment (sthirugn) [inline tests/foreman/ui/test_variables.py]: s/smartVariables/smart_variables?

File: tests/foreman/ui/test_variables.py [hunk 2]
```diff
@@ -0,0 +1,1130 @@
...
+        @assert: The Merge Overrides, Merge Default checkboxes
+        are not enabled to check.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_positive_enable_avaoid_duplicates_checkbox(self):
```
Comment (sthirugn) [inline tests/foreman/ui/test_variables.py]: s/avaoid/avoid

File: tests/foreman/ui/test_variables.py [hunk 3]
```diff
@@ -0,0 +1,1130 @@
...
+        @feature: Smart Variables - Create
+
+        @steps:
+
+        1. Create a smart Variable with Valid name and default value.
+
+        @assert:
+
+        1. The smart Variable is created successfully.
+        2. In YAML output of associated host, the variable with name and and
```
Comment (oshtaier) [inline tests/foreman/ui/test_variables.py]: `and and` typo

File: tests/foreman/ui/test_variables.py [hunk 4]
```diff
@@ -0,0 +1,1130 @@
...
+        its respective puppet class.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_negative_create_variable(self):
+        """Smart Variable is not created with invalid details.
```
Comment (oshtaier) [inline tests/foreman/ui/test_variables.py]: maybe `with invalid data.`

File: tests/foreman/ui/test_variables.py [hunk 5]
```diff
@@ -0,0 +1,1130 @@
...
+
+        @feature: Smart Variables - Optional Validation
+
+        @steps:
+
+        1.  Create a variable with empty value.
+        2.  Check Required checkbox in 'Optional Input Validator'.
+        3.  Submit the change.
+
+        @assert: Error raised for blank default value by 'Required' checkbox.
```
Comment (oshtaier) [inline tests/foreman/ui/test_variables.py]: `Error is raised`
For all cases

File: tests/foreman/ui/test_variables.py [hunk 6]
```diff
@@ -0,0 +1,1130 @@
...
+
+        @feature: Smart Variables - Optional Validation
+
+        @steps:
+
+        1.  Provide some default value.
+        2.  Check Required checkbox in 'Optional Input Validator'.
+        3.  Submit the change.
+
+        @assert: Error not raised default value by 'Required' checkbox.
```
Comment (oshtaier) [inline tests/foreman/ui/test_variables.py]: `Error is not raised for default value`
For all cases

File: tests/foreman/ui/test_variables.py [hunk 7]
```diff
@@ -0,0 +1,1130 @@
...
+        @assert: Error not raised for matcher value by 'Required' checkbox.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_negative_validate_default_value_with_regex(self):
+        """Error raised for default value not matching with regex.
```
Comment (oshtaier) [inline tests/foreman/ui/test_variables.py]: `Error is raised for default value which is not matching with regex.`

File: tests/foreman/ui/test_variables.py [hunk 8]
```diff
@@ -0,0 +1,1130 @@
...
+    @stubbed()
+    @tier3
+    def test_negative_validate_matcher_value_with_regex(self):
+        """Error raised for matcher value not matching with regex.
+
+        @feature: Smart Variables - Optional Validation
+
+        @steps:
+
+        1.  Create a matcher with value that doesn't matches
```
Comment (oshtaier) [inline tests/foreman/ui/test_variables.py]: `doesn't match`

File: tests/foreman/ui/test_variables.py [hunk 9]
```diff
@@ -0,0 +1,1130 @@
...
+    @tier3
+    def test_negative_validate_matcher_value_with_default_type(self):
+        """Error raised for matcher value not of default type.
+
+        @feature: Smart Variables - Validation
+
+        @steps:
+
+        1.  Update variable default type with valid value.
+        2.  Create a matcher with value that doesn't matches the default type.
```
Comment (oshtaier) [inline tests/foreman/ui/test_variables.py]: `doesn't match`
And for all cases

**Comment by jyejare** [comment]
@sthirugn , @oshtaier , Ping ?

File: tests/foreman/ui/test_variables.py [hunk 10]
```diff
@@ -0,0 +1,1131 @@
...
+        checkbox.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_positive_validate_default_value_required_checkbox(self):
+        """Error not raised for default Value - Required checkbox.
```
Comment (oshtaier) [inline tests/foreman/ui/test_variables.py]: please fix
```
Error is not raised
```
and 
```
Error is raised
```
for all cases

File: tests/foreman/ui/test_variables.py [hunk 11]
```diff
@@ -0,0 +1,1131 @@
...
+from robottelo.test import UITestCase
+
+
+class SmartVariablesTestCase(UITestCase):
+    """Implements Smart Variables tests in UI"""
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_create_variable(self):
```
Comment (sthirugn) [inline tests/foreman/ui/test_variables.py]: test_positive_create?  since this is already in test_variables.py

**Comment by sthirugn** [comment]
ACK pending minor comment

**Comment by jyejare** [comment]
@oshtaier and @sthirugn I updated the PR as per your comments.

**Comment by sthirugn** [comment]
ACK

**Comment by oshtaier** [comment]
ACK

**Pull request merged**